Sources/SwiftSyntaxBuilderGeneration/Templates/ResultBuildersFile.swift

```diff
@@ -0,0 +1,387 @@
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+let resultBuildersFile = SourceFile {
+  ImportDecl(
+    leadingTrivia: .docLineComment(copyrightHeader),
+    path: "SwiftSyntax"
+  )
+  for node in SYNTAX_NODES where node.isSyntaxCollection {
+    let type = SyntaxBuildableType(syntaxKind: node.syntaxKind)
+    let elementType = node.collectionElementType
+    
+    StructDecl(
+      attributes: TokenSyntax.identifier("@resultBuilder").withLeadingTrivia(.newlines(1)).withTrailingTrivia(.newlines(1)),
+      modifiers: TokenSyntax.public,
+      identifier: type.syntaxKind + "Builder") {
+        TypealiasDecl(
+          leadingTrivia: [
+            .newlines(1),
+            .docLineComment("/// The type of individual statement expressions in the transformed function,"),
+            .newlines(1),
+            .docLineComment("/// which defaults to Component if buildExpression() is not provided."),
+            .newlines(1)
+          ],
+          modifiers: [TokenSyntax.public],
+          identifier: "Expression",
+          initializer: elementType.expressibleAs)
+        
+        TypealiasDecl(
+          leadingTrivia: [
+            .newlines(1),
+            .docLineComment("/// The type of a partial result, which will be carried through all of the"),
+            .newlines(1),
+            .docLineComment("/// build methods."),
+            .newlines(1)
+          ],
+          modifiers: [TokenSyntax.public],
+          identifier: "Component",
+          initializer: ArrayType(elementType: elementType.expressibleAs))
+        
+        TypealiasDecl(
+          leadingTrivia: [
+            .newlines(1),
+            .docLineComment("/// The type of the final returned result, which defaults to Component if"),
+            .newlines(1),
+            .docLineComment("/// buildFinalResult() is not provided."),
+            .newlines(1)
+          ],
+          modifiers: [TokenSyntax.public],
+          identifier: "FinalResult",
+          initializer: type.buildable)
+        
+        
+        //      public static func buildBlock(_ components: Component...) -> Component {
+        //        return components.flatMap { $0 }
+        //      }
+        FunctionDecl(
+          leadingTrivia: [
+            .newlines(1),
+            .docLineComment("/// Required by every result builder to build combined results from"),
+            .newlines(1),
+            .docLineComment("/// statement blocks."),
+            .newlines(1)],
+          modifiers: [TokenSyntax.public, TokenSyntax.static],
+          identifier: .identifier("buildBlock"),
+          signature: FunctionSignature(
+            input: ParameterClause {
+              FunctionParameter(
+                firstName: TokenSyntax.wildcard,
```

Comment:
`TokenSyntax` can be omitted in parameters that only expect `TokenSyntax`, i.e. we can use implicit member expressions, e.g. `.wildcard` here.